.github/workflows/helm.yml

```diff
@@ -0,0 +1,224 @@
+
+on:
+  workflow_call:
+    inputs:
+        charts-path:
+            type: string
+            required: false
+            description: The path of the Charts directory, defaults to charts/<repo>
+        charts-name:
+            type: string
+            description: The name of charts, defaults to {{ github.repository }} <owner>/<repository>
+            default: ${{ github.repository }}
+            required: false
+        path:
+            type: string
+            description: The path for the output folder of clone
+            default: ${{ github.repository_id }}
+            required: false
+        helm-repositories:
+            description: 'JSON-encoded list of Helm repositories.'
+            required: true
+            type: string
+
+jobs:
+  prepare-charts-path:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Set default charts-path if not provided
+      id: set-path
+      run: |
+        REPO_NAME=$(echo ${{ github.repository }} | cut -d'/' -f2)
+        echo "repo=$REPO_NAME" >> $GITHUB_OUTPUT
+        if [ -z "${{ inputs.charts-path }}" ]; then
+          echo "path=./charts/$REPO_NAME" >> $GITHUB_OUTPUT
+        else
+          echo "path=${{inputs.charts-path}}" >> $GITHUB_OUTPUT
+        fi
+    outputs:
+      charts_path: ${{ steps.set-path.outputs.path }}
+      repository_name: ${{ steps.set-path.outputs.repo }}
+  lint:
+    name: Lints
+    runs-on: ubuntu-latest
+    needs: prepare-charts-path
+    steps:
+        - uses: actions/checkout@v4
+        - uses: azure/setup-helm@v3.5
+          with:
+            version: "v3.12.0"
+          id: install
+        - name: Build dependencies
+          run: helm dep update ${{ needs.prepare-charts-path.outputs.charts_path }}
+        - name: Lint Chart
+          run: helm lint ${{ needs.prepare-charts-path.outputs.charts_path }}
+  test-if-version-changed:
+    name: Tests if the Chart version is updated and CHANGELOG is updated for the version
+    runs-on: ubuntu-latest
+    needs: prepare-charts-path
+    steps:
+      - uses: actions/checkout@v4
+      - uses: azure/setup-helm@v3.5
+        with:
+          version: "v3.12.0"
+        id: install
+
+      - name: Install tools
+        run: sudo snap install yq
+      - name: Get Chart version
+        run: |
+          echo CHART_VERSION=$(yq eval .version ${{ needs.prepare-charts-path.outputs.charts_path }}/Chart.yaml) >> $GITHUB_OUTPUT
+        id: get-chart-version
+      - name: Validate version increment
+        run: |
+          helm repo add substra https://substra.github.io/charts
+
+          RES=$(helm search repo ${{ inputs.charts-name }} --version ${{ steps.get-chart-version.outputs.CHART_VERSION }})
+
+          if [ "${RES}" == "No results found" ]; then
+            echo "Version incremented"
+          else
+            echo "Validation failed, you should upgrade the chart version in Chart.yaml"
+            exit 1
+          fi
+      - name: Check for version and date in CHANGELOG
```

Comment:
This is a good idea but I worry it will be a bit fragile because we failed to standardize changelog formats.